src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -208,6 +245,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
                 }
             }
         }
+        if (emojiFlag.equals(tokenizedMessageContent.get(i))){
+            String shortcode = "";
+            Boolean validSyntax = false;
+            int j;
+            for (j = i+1; j < tokenizedMessageContent.size(); j++){
```

Comment:
I would suggest using the two-argument `String.indexOf()` and `String.substring()` to be more explicit

Reply:
I don't think there's an elegant way to do this using substring, because each character is still considered a separate token. Maybe entire emoji "shortcode" blocks should be treated as a single token...